scripts/normalize-by-median.py

```diff
@@ -33,188 +33,114 @@
 import argparse
 from khmer.kfile import (check_space, check_space_for_hashtable,
                          check_valid_file_exists)
-from khmer.utils import write_record, check_is_pair
+from khmer.utils import write_record, check_is_pair, broken_paired_reader
 DEFAULT_DESIRED_COVERAGE = 10
 
-# Iterate a collection in arbitrary batches
-# from: http://stackoverflow.com/questions/4628290/pairs-from-single-list
 
-
-def batchwise(coll, size):
-    iter_coll = iter(coll)
-    return izip(*[iter_coll] * size)
-
-# Returns true if the pair of records are properly pairs
-
-
-def WithDiagnostics(ifile, batch_size, fp, paired, norm):
+def WithDiagnostics(ifilename, norm, reader, fp):
     """
-    Generator/context manager to do boilerplate output of statistics while
-    normalizing data. Also checks for properly paired data.
+    Generator/context manager to do boilerplate output of statistics using a
+    Normalizer object.
     """
 
     index = 0
 
-    for index, batch in enumerate(batchwise(
-                                  screed.open(ifile, parse_description=False),
-                                  batch_size)):
+    # per read diagnostic output
+    for index, record in enumerate(norm(reader)):
 
-        norm.total += batch_size
-        total = norm.total
-        discarded = norm.discarded
-
-        if index > 0 and index % 100000 == 0:
-            print('... kept {kept} of {total} or {perc:2}%'
-                  .format(kept=total - discarded,
-                          total=total,
-                          perc=int(100. - discarded / float(total) * 100.)),
+        if norm.total > 0 and norm.total % 10000 == 0:
```

Comment:
Note: seeing as you've implemented a specific diagnostics handler, you
*could* parameterize this... :)
On Jun 7, 2015 8:13 PM, "Jake Fenton" <notifications@github.com> wrote:

> In scripts/normalize-by-median.py
> <https://github.com/dib-lab/khmer/pull/1057#discussion_r31884351>:
>
> >
> > -        norm.total += batch_size
> > -        total = norm.total
> > -        discarded = norm.discarded
> > -
> > -        if index > 0 and index % 100000 == 0:
> > -            print('... kept {kept} of {total} or {perc:2}%'
> > -                  .format(kept=total - discarded,
> > -                          total=total,
> > -                          perc=int(100. - discarded / float(total) * 100.)),
> > +        if norm.total > 0 and norm.total % 10000 == 0:
>
> Ahh, excellent. I don't have to worry about it, then.
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/dib-lab/khmer/pull/1057/files#r31884351>.
>